### DISCUSSION TITLE
New RS458 adapter

### DISCUSSION THREAD

**Discussion by mzuniga51** [tagyoureit/nodejs-poolController#767] [App and O/S Config Questions]
Hi @rstrouse.  Hope you are doing well.  I have purchased a new rs458 adapter connected via LAN in the hopes of making my setup more stable as the old ones working via wifi were quite unstable.  So this is the adapter I bought https://www.amazon.com/dp/B07G56LBQJ?psc=1&ref=ppx_yo2ov_dt_b_product_details which has the two rs458 ports I need and it is connected to my network via Lan. I hooked it up and configure it like this:
<img width="922" alt="Screenshot 2023-05-26 at 09 08 34" src="https://github.com/tagyoureit/nodejs-poolController/assets/91519307/fdeb4560-bf83-4dbf-bbf9-fdda5b95d2ed">
<img width="870" alt="Screenshot 2023-05-26 at 08 57 45" src="https://github.com/tagyoureit/nodejs-poolController/assets/91519307/8c769edb-b775-40e2-a492-de671c36928c">
<img width="890" alt="Screenshot 2023-05-26 at 09 09 11" src="https://github.com/tagyoureit/nodejs-poolController/assets/91519307/c7b14533-792e-409c-985e-b15d51416e54">
I made the auxiliary port work on the njsPC work by deleting it and setting it up again and I could see packets going in and out from the Intellicenter so that makes me believe the adapter wires are correctly connected.  The main port however didn't worked.  I tried then resetting it by deleting the Intellichlor associated with it in the njsPC but were unable to do so.  So I tried to update njsPC to see if this might help but unfortunately I then lost all connection to the adapter and obviously to the Intellicenter itself.  I receive this message 
<img width="722" alt="Screenshot 2023-05-26 at 09 01 10" src="https://github.com/tagyoureit/nodejs-poolController/assets/91519307/10732685-7a21-456f-9122-909238f148b6">
I would be very grateful if you could point me in the right direction as to what may be happening here.  Please let me know if you need me to send any logs.   All the best and as always many thanks!

**Comment by tagyoureit** [comment]
If you click close and then save a second time, does it save the port properly?

**Reply by mzuniga51** [reply]
No. Same message.

**Reply by mzuniga51** [reply]
Sorry, sorry. It did work!  Now I have the two ports open.  Still no data flow from intellicenter though.  let me restart the adapter and see what happens
<img width="651" alt="Screenshot 2023-05-26 at 10 37 05" src="https://github.com/tagyoureit/nodejs-poolController/assets/91519307/38a38c94-44f5-45a9-a250-e0f74f487e11">

**Comment by rstrouse** [comment]
Double check your TCP port your screenshots first show 23 then they show 26.  If it is now 26 you need to change that in dashPanel.

EDIT: Make the changes then restart njsPC.  Perhaps we block the start when the port closure fails.  It won't try to instantiate the incorrect port on a restart.

**Reply by mzuniga51** [reply]
Ports are correct.  23 for main and 26 for aux.  I just happened to send the wrong pic. Sorry!

**Reply by mzuniga51** [reply]
@tagyoureit suggestion worked and ports were created.  I restarted everything but now it is down for some reason.  Still investigating.

**Comment by mzuniga51** [comment]
Ok, so now the adapter is connected 
![Screenshot 2023-05-26 at 11 18 17](https://github.com/tagyoureit/nodejs-poolController/assets/91519307/df1fba6f-e3ca-4316-a740-39dae26d592e)
And the ports are open
![Screenshot 2023-05-26 at 11 18 57](https://github.com/tagyoureit/nodejs-poolController/assets/91519307/6d780c31-3b88-42a8-9077-ee78bed6fd6d)
Aux port seem to be receiving data but principal port is dry.  Still no reading config from Intellicenter though.  Any ideas?  I've reset the server (MacMini) and the adapter.  Only thing I haven done is reset the Intellicenter itself.

**Reply by mzuniga51** [reply]
Main port received some info two now.  Still not reading the data from the Intellicenter

**Reply by rstrouse** [reply]
Not sure what is on the aux port but it looks like it is garbage.  What should be communicating on that port?  njsPC hasn't asked for anything so it should be silent.

**Reply by mzuniga51** [reply]
It should read the pool intellichlor

**Reply by rstrouse** [reply]
Not if an IntelliChlor was not defined in the configuration attached to that port.  It should be silent until Nixie says `Hey IntelliChlor I am in control now`

**Reply by mzuniga51** [reply]
Yes I figured as much.  So why is not Nixie taking control?  Or maybe the question is why njcPC is not reading the configuration of the Intellicenter?  What im I missing?

**Reply by rstrouse** [reply]
Does that allow socket B to operated as a TCP server?  This seems to suggest that only one of the ports will work as a TCP server.

![image](https://github.com/tagyoureit/nodejs-poolController/assets/47839015/e845a417-7fb4-4779-88ac-013d2e3f62b8)

**Reply by rstrouse** [reply]
Oh I get it Socket A/B are not the ports.  Each port has two sockets associated with them which I don't understand that utility unless they expect to share with a read only second connection.

**Reply by mzuniga51** [reply]
Correct. So, stupid me I found I had the cables to the adapter connected wrong (pinout is not the most intuitive between the serial male and the female on the other side) so now dashPanel identifies my Control Panel correctly as Intellicenter i10D.  However now I cannot save the ports again
![Screen Shot 2023-05-26 at 13 03 01](https://github.com/tagyoureit/nodejs-poolController/assets/91519307/b348268e-b3b8-4097-932b-18b981f98477)

**Reply by rstrouse** [reply]
A couple of actions to take.

* Reserve the IP address for your adapter in DHCP and make sure it stays the same
* Restart njsPC

**Reply by mzuniga51** [reply]
Yeah it is reserved.  Nothing

**Reply by rstrouse** [reply]
Nothing as in cannot save the port or it is open but not receiving.

**Comment by mzuniga51** [comment]
Sorry.  What do you mean?

**Reply by rstrouse** [reply]
You said...
> Yeah it is reserved. Nothing

I was asking what do you mean?

**Comment by mzuniga51** [comment]
Ohh ok.  That the njsPC en is still not working.  I dont understand why i
cannot add the two ports.  I think now is propery connected.

**Reply by rstrouse** [reply]
I think I am not understanding what you mean by the `njsPC en` is still not working.
